src/common/config/environment.config.ts

```diff
@@ -1,3 +1,91 @@
-export default class Environment {
+const env = process.env.NODE_ENV || "local";
 
-}
\ No newline at end of file
+if (env === "development") {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const dotenv = require("dotenv");
+  dotenv.config();
+}
+
+export interface Environment {
+  dbconf: DBConfig;
+  hostconf: HostConfig;
+}
+
+export interface HostConfig {
+  host: string;
+  port: number;
+}
+
+export interface DBConfig {
+  user: string;
+  password: string;
+  url: string;
+  port: number;
+}
+
+// used for local development
+const devConf: Environment = {
+  dbconf: {
+    user: process.env.LOCAL_DBUSER || "",
+    password: process.env.LOCAL_DBPASS || "",
+    url: process.env.LOCAL_DBURL || "",
+    port: Number(process.env.LOCAL_DBPORT),
+  },
+
+  hostconf: {
+    host: process.env.LOCAL_HOST || "",
+    port: Number(process.env.LOCAL_PORT || ""),
+  },
+};
+
+const stagingConf: Environment = {
```

Comment:
I think setting up a staging environment now would be over engineering, we can have more than enough with dev, testing and production imo.

Reply:
For visibility: we agreed on having the staging environment variables be for the CI/CD pipeline.